src/ds1302.c

```diff
@@ -144,39 +136,57 @@ void ds_reset_clock() {
 }
     
 void ds_hours_12_24_toggle() {
-    uint8_t b = 0;
-    b = (! rtc.h24.hour_12_24) << 7;    // toggle 12/24 bit
-    ds_writebyte(DS_ADDR_HOUR, b);
+
+    uint8_t hours,b;
+    if (H12_24)
+    { // 12h->24h
+      hours=ds_split2int(rtc_table[DS_ADDR_HOUR]&0x1F); //12h format (1-11am 12pm 1-11pm 12am)
```

Comment:
the bit masks here for the elements of the rtc_table here might be unfamiliar to the reader. I wonder if making these masks a symbol (macro constant), e.g.
#define   DS_ADDR_HOUR_MASK   0b00011111 
for portability within the code would help. Yes, it would be more verbose code, but for easier for me (and hopefully other beginners experimenting with the code) to read and understand.